### DISCUSSION TITLE
Combine string and variables for url template

### DISCUSSION THREAD

**Discussion by asportnoy** [JanDeDobbeleer/oh-my-posh#3168] [Troubleshoot]
I'm trying to use the URL template to get a link to the current branch on the upstream. `{{ url .HEAD .UpstreamURL }}` works fine, but I'm struggling to figure out how to append `/tree/{{ .Ref }}` to it. My best guess is `{{ url .HEAD "{{ .UpstreamURL | replace ".git" "" }}/tree/{{ .Ref }}" }}` but that gives the error "unable to create text based on template" and I can't quite figure out the correct syntax.

I was able to manually create the link by putting `\e]8;;{{ .UpstreamURL | replace ".git" "" }}/tree/{{ .Ref }}\e\\{{ .HEAD }}\e]8;;\e\\` but I'd prefer to use the template if possible. Any help is appreciated!

**Comment by JanDeDobbeleer** [comment]
You'll need to create a variable for this.

```
{{ $link := print (.UpstreamURL | replace \".git\" \"\") \"/tree/\" .Ref }}{{ url .HEAD $link }}
```

**Reply by asportnoy** [reply]
@JanDeDobbeleer This gives me "invalid template text". I tried removing the URL block and just putting `{{ $link }}` but it persists.

**Reply by JanDeDobbeleer** [reply]
@asportnoy I have this one right now which works:

```json
"template": " {{ $link := print (.UpstreamURL | replace \".git\" \"\") \"/tree/\" .Ref }}{{ url .HEAD $link }} ",
```

<img width="1011" alt="image" src="https://user-images.githubusercontent.com/2492783/205994700-d2a34424-d402-4c6b-8dc3-5134e5ee2f20.png">
